api/src/main/java/com/nicknathanjustin/streamercontracts/security/SecurityServiceImpl.java

```diff
@@ -1,21 +1,66 @@
 package com.nicknathanjustin.streamercontracts.security;
 
+import com.nicknathanjustin.streamercontracts.users.UserModel;
+import com.nicknathanjustin.streamercontracts.users.UserService;
 import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.security.oauth2.provider.OAuth2Authentication;
 
 import javax.servlet.http.HttpServletRequest;
 import java.security.Principal;
+import java.util.List;
 
+@RequiredArgsConstructor
+@Slf4j
 public class SecurityServiceImpl implements SecurityService {
 
+    @NonNull private final UserService userService;
+
     @Value("${twitch.extension.jwtHeader}")
     private String jwtHeader;
 
+    @Value("${twitch.whiteListedAccounts}")
+    private List<String> whiteListedAccounts;
+
+    @Value("${application.blackListedAccounts}")
+    private List<String> blackListedAccounts;
+
+    @Value("${application.environment}")
+    private String environment;
+
     @Override
     public boolean isAnonymousRequest(@NonNull final HttpServletRequest httpServletRequest) {
         final String jwtToken = httpServletRequest.getHeader(jwtHeader);
         final Principal principal = httpServletRequest.getUserPrincipal();
-        return jwtToken == null && !(principal instanceof OAuth2Authentication);
+        final boolean isUserRecognized = (jwtToken != null || principal instanceof OAuth2Authentication);
+
+        if (isUserRecognized) {
+            return isRecognizedUserBlocked(httpServletRequest);
+        }
+        return true;
+    }
+
+    private boolean isRecognizedUserBlocked(@NonNull final HttpServletRequest httpServletRequest) {
+        final UserModel userModel = userService.getUserModelFromRequest(httpServletRequest);
+        if (isEnvLockedToWhiteListedAccounts()) {
+            return !isUserWhiteListed(userModel);
+        }
+        return isUserBlackListed(userModel);
+    }
+
+    private boolean isEnvLockedToWhiteListedAccounts() {
+        // Beta endpoint is currently open to any client. However, we want to only allow bountyStreamer accounts to
```

Comment:
Whitelist is for beta, blacklist is for prod?

Reply:
yiss